cpp/cmake/Modules/ConfigureArrow.cmake

```diff
@@ -47,48 +66,33 @@ else()
     set(PARALLEL_BUILD --parallel)
     message("STATUS Enabling Parallel CMake build")
 endif($ENV{TRAVIS})
-
+ 
 execute_process(
-    COMMAND ${CMAKE_COMMAND} --build ${PARALLEL_BUILD} ..
+    COMMAND ${CMAKE_COMMAND} --build ${PARALLEL_BUILD} .
     RESULT_VARIABLE ARROW_BUILD
-    WORKING_DIRECTORY ${ARROW_ROOT}/build)
+    WORKING_DIRECTORY ${ARROW_ROOT}/download)
 
 if(ARROW_BUILD)
-    message(FATAL_ERROR "Building Arrow failed: " ${ARROW_BUILD})
-endif(ARROW_BUILD)
-
-set(ARROW_GENERATED_IPC_DIR 
-    "${ARROW_ROOT}/build/src/arrow/ipc")
-
-configure_file(${ARROW_GENERATED_IPC_DIR}/File_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/File_generated.h COPYONLY)
-configure_file(${ARROW_GENERATED_IPC_DIR}/Message_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/Message_generated.h COPYONLY)
-configure_file(${ARROW_GENERATED_IPC_DIR}/Schema_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/Schema_generated.h COPYONLY)
-configure_file(${ARROW_GENERATED_IPC_DIR}/Tensor_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/Tensor_generated.h COPYONLY)
+    message(FATAL_ERROR "Build step for Arrow failed: ${ARROW_BUILD}")
+endif()
 
-message(STATUS "Arrow installed here: " ${ARROW_ROOT}/install)
-set(ARROW_LIBRARY_DIR "${ARROW_ROOT}/install/lib")
-set(ARROW_INCLUDE_DIR "${ARROW_ROOT}/install/include")
-
-find_library(ARROW_LIB arrow
-             NO_DEFAULT_PATH
-             HINTS "${ARROW_LIBRARY_DIR}")
-
-if(ARROW_LIB)
-    message(STATUS "Arrow library: " ${ARROW_LIB})
-    set(ARROW_FOUND TRUE)
-endif(ARROW_LIB)
-
-set(FLATBUFFERS_ROOT "${ARROW_ROOT}/build/flatbuffers_ep-prefix/src/flatbuffers_ep-install")
 
+# Add transitive dependency: Flatbuffers
+set(FLATBUFFERS_ROOT ${ARROW_ROOT}/build/flatbuffers_ep-prefix/src/flatbuffers_ep-install/)
 message(STATUS "FlatBuffers installed here: " ${FLATBUFFERS_ROOT})
 set(FLATBUFFERS_INCLUDE_DIR "${FLATBUFFERS_ROOT}/include")
 set(FLATBUFFERS_LIBRARY_DIR "${FLATBUFFERS_ROOT}/lib")
 
 add_definitions(-DARROW_METADATA_V4)
-add_definitions(-DARROW_VERSION=1000)
-
-
-
+add_definitions(-DARROW_VERSION=1101) # DARROW_VERSION=1101 is related to apache-arrow-0.11.1
```

Comment:
Our library builds with `0.10.0` by default.

I think it makes sense to turn this into a CMake option rather than hard-coding.

Reply:
@mt-jones  So this PR is to upgrade to arrow-0.11.1. We will figure out and use c cmake option to specify arrow version  which are at least 0.10.0. @aucahuasi